libraries/vault_installation_binary.rb

```diff
@@ -82,14 +82,17 @@ def vault_program
       end
 
       def self.default_archive_url
-        "https://releases.hashicorp.com/vault/%{version}/%{basename}" # rubocop:disable Style/StringLiterals
+        "https://%{resource.archive_url_root}/vault/%{version}/%{basename}" # rubocop:disable Style/StringLiterals
```

Comment:
I can't put this on like 36 like I want, but I think this doesn't quite work, since this function doesn't have access to `resource`. You have two ways to do this:

1) In `default_inversion_options` on L36 you can specify a new variable that is interpolated (this is just a string that gets formatted later on) and update the default archive url.
2) You can extend `default_archive_url` and pass in the `resource` object where the method is called such that the new `archive_url_root` policy is available in scope.

I think the former aligns more with the intent of this method but the latter is also workable.